hr_holidays_leave_auto_approve/models/hr_holidays.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# © 2016 ONESTEiN BV (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api
+
+
+class HrHolidays(models.Model):
+    _inherit = "hr.holidays"
+
+    @api.model
+    @api.returns('self', lambda value: value.id)
+    def create(self, values):
+        result = super(HrHolidays, self).create(values)
+        if result.holiday_status_id.auto_approve:
+            result.sudo().signal_workflow('validate')
```

Comment:
Just overriding the `_check_state_access_right` to build some logic dependent on `holiday_status_id` is not going to work, as this method is called also in the `write` and usually we don't have access to `holiday_status_id` into `vals`, when called by a `write`.
Now, we are sure that during the process of auto approval the write is called several times and in particular when the state is changed by the workflow transition. When this happens, as we have `state` into `values` and we don't have any way to `access holiday_status_id`, in case we're using a user without `base.group_hr_user group,` the system will always raise the error.

In addition, in my opinion it's a conceptual mistake to use the requesting user to validate: it's not really the user that is validating the request, the system does it. For me it's much more coherent that fields such as `write_uid`, but especially the state-change track message, report that was the system to approve the request and not the user himself.

In my opinion, we should rather look at this functionality like at a kind of automatic action with immediate trigger upon the creation. Let's imagine that instead than implementing the auto-approval functionality into the create we decided to configure a cron job in order to change the state of those leave requests, not approved yet, that belongs to an auto-approval leave type. In this case the write would be done by the admin user and it would be considered correct.

Following these thoughts I consider more correct to solve the issue incurring when the admin user doesn't have the `base.group_hr_user group` as I did in my latest commit. Please, give a look at it.